.github/workflows/ci.yml

```diff
@@ -57,65 +65,70 @@ jobs:
         with:
-          name: consul-logs-${{ matrix.os }}
-          path: Consul.Test/consul*.log
-      - name: Create NuGet package
-        if: runner.os == 'Windows'
-        run: |
-          if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
-            # WORKAROUND: Add letter prefix to ensure that MSBuild treats
-            # the suffix as a string. e.g. '0313071' will fail as an invalid
-            # version string, but 'G0313071' will succeeded. It looks like
-            # the parser does not like numbers with a leading zero.
-            $suffix = 'g' + $(git rev-parse --short HEAD)
-            $params = "--version-suffix", $suffix
-          }
+          name: consul-logs-${{ matrix.consul }}-${{ matrix.framework }}-${{ matrix.os }}
+          path: consul*.log
 
-          dotnet pack --configuration=Release --no-build @params Consul/Consul.csproj
+  package:
+    runs-on: windows-2019
+    needs: [Consul_AspNetCore, Consul]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Create NuGet package
+      run: |
+        if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
+          # WORKAROUND: Add letter prefix to ensure that MSBuild treats
+          # the suffix as a string. e.g. '0313071' will fail as an invalid
+          # version string, but 'G0313071' will succeeded. It looks like
+          # the parser does not like numbers with a leading zero.
+          $suffix = 'g' + $(git rev-parse --short HEAD)
+          $params = "--version-suffix", $suffix
+        }
+
+        dotnet pack --configuration=Release --output dist @params
 
-          if ("${{ github.ref }}" -like "refs/tags/v*") {
+        if ("${{ github.ref }}" -like "refs/tags/v*") {
             $tag = "${{ github.ref }}".SubString(11)
-            $expectedFile = "dist/consul/Consul.$tag.nupkg"
+            $expectedFile = "dist/Consul.$tag.nupkg"
             # Check whether the tag and the package version match together
             if (-not (Test-Path -Path $expectedFile)) {
                 echo "Expected file $expectedFile doesn't exist"
                 exit 1
             }
-          }
-
-      - name: Upload NuGet package artifact
-        if: runner.os == 'Windows'
-        uses: actions/upload-artifact@v2
-        with:
-          name: nuget-package
-          path: dist/consul
+        }
+    - name: Upload NuGet package artifact
+      uses: actions/upload-artifact@v2
+      with:
+        path: dist/*.nupkg
 
   # Publish NuGet package when a tag is pushed.
-  # Tests need to succeed on all platforms first, including having a tag name that matches the version number.
+  # Tests need to succeed for all components and on all platforms first,
+  # including having a tag name that matches the version number.
   publish:
-    if: startsWith(github.ref, 'refs/tags/v')
-    needs: build
-    runs-on: ubuntu-18.04
+    if: github.ref == 'refs/heads/master' && startsWith(github.ref, 'refs/tags/v')
```

Comment:
I don't think this will ever get triggered - any reason you changed this condition?

Reply:
> I don't think this will ever get triggered

Why not? It only checks that we are on master.

> any reason you changed this condition from the previous?

Just an extra defensive check to avoid this: https://github.com/mfkl/consuldotnet/actions/runs/967076197

If someone or something creates pushes a new branch to this repo, and tags it, the CI would try to deploy to NuGet. I can remove it, if you prefer.

Reply:
I think this won't work because `github.ref` cannot both be a branch and a tag. Pushing to a branch will trigger a workflow, and pushing to a tag will trigger a different one.

In my opinion, we should just go for the same check as before.

Then in the future we should start using GitHub Actions Environments and the `workflow_run` event to ensure only a tag on `master` can push to a package to NuGet. This is something that our DevOps team could add to this repo.

Reply:
Fair enough.